### PR TITLE
partner: Pinecone - Update Poetry requirements for pinecone-client >=3.2.2

### PR DESCRIPTION
**Pull request by cwaddingham**
This change updates the requirements in `libs/partners/pinecone/pyproject.toml` to allow all versions of `pinecone-client` greater than or equal to 3.2.2.

This change resolves issue [21955](https://github.com/langchain-ai/langchain/issues/21955).

### CODE REVIEW

File: libs/partners/pinecone/pyproject.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ license = "MIT"
 # <3.13 is due to restriction in pinecone-client package
 python = ">=3.8.1,<3.13"
 langchain-core = ">=0.1.52,<0.3"
-pinecone-client = "^3.2.2"
+pinecone-client = ">=3.2.2"
```
Comment (efriis) [inline libs/partners/pinecone/pyproject.toml:17]:
```suggestion
pinecone-client = ">=3.2.2,<5"
```
should we cap it at 5 in the event that major release is a breaking change?

File: libs/partners/pinecone/tests/unit_tests/test_vectorstores.py [hunk 2]
```diff
@@ -10,3 +10,15 @@ def test_initialization() -> None:
...
+    """Test integration of the id_prefix parameter."""
+    embedding = Mock()
+    embedding.embed_documents = Mock(return_value=[0.1,0.2,0.3,0.4,0.5])
+    index = Mock()
+    index.upsert = Mock(return_value=None)
+    text_key = "testing"
+    vectorstore = PineconeVectorStore(index, embedding, text_key)
+    texts = ["alpha", "beta", "gamma", "delta", "epsilon"]
+    id_prefix = "testing_prefixes"
+    vectorstore.add_texts(texts, id_prefix=id_prefix, async_req=False)
```
Comment (efriis) [inline libs/partners/pinecone/tests/unit_tests/test_vectorstores.py:24]:
```suggestion
    vectorstore.add_texts(texts, id_prefix=id_prefix, async_req=False)

```

**Review by efriis — COMMENTED** [review]
two nits, otherwise lgtm

**Comment by cwaddingham** [comment]
Nits look, accepted both.

**Review by baskaryan — APPROVED** [review]

**Comment by walterheck** [comment]
@baskaryan I think the issues are small, can you check? Anything I can do?

**Comment by walterheck** [comment]
@baskaryan sorry for bothering you once more :)

**Review by ccurme — APPROVED** [review]

**Pull request merged**